tools/mo/unit_tests/mo/middle/permute_tensor_iterator_test.py

```diff
@@ -0,0 +1,137 @@
+
+        ti_node = Node(graph, 'ti')
+        ti_node.body = sub_graph
+
+        TransposeTensorIteratorLSTM().find_and_replace_pattern(graph)
+        graph.clean_up()
+
+        graph_ref = build_graph(nodes,
+                                [('input', 'input_d'),
+                                 ('input_d', 'ti', {'in': 0, 'external_port_id': 0}),
+                                 *connect('init_hidden', '1:ti'),
+                                 *connect('init_cell', '2:ti'),
+                                 ('ti', 'ti_d', {'external_port_id': 0}),
+                                 ('ti_d', 'output', {'in': 0}), ],
+                                {'input_d': {'shape': [3, 1, 10]},
+                                 'ti': {'output_port_map': [{'axis': 0, 'start': -1, 'end': 0, 'stride': -1,
+                                                             'external_port_id': 0}],
+                                        'input_port_map': [{'axis': 0, 'start': 0, 'end': -1, 'stride': 1,
+                                                            'external_port_id': 0}]},
+                                 'ti_d': {'shape': [1, 3, 10]}}, nodes_with_edges_only=True)
+
+        sub_graph_ref = build_graph(nodes_internal,
+                                    [*connect('input_unsqueezed', '0:squeeze'),
+                                     *connect('squeeze_dim', '1:squeeze'),
+                                     *connect('squeeze', '0:lstm'),
+                                     *connect('input_hidden', '1:lstm'),
+                                     *connect('input_cell', '2:lstm'),
+                                     *connect('weights', '3:lstm'),
+                                     *connect('biases', '4:lstm'),
+                                     *connect('lstm:0', 'unsqueeze'),
+                                     ('lstm', 'lstm_d2', {'out': 1}),
+                                     *connect_data('lstm:0', 'op_output_1'),
+                                     ('lstm_d2', 'op_output_2'),
+                                     *connect('unsqueeze_dim', '1:unsqueeze'),
+                                     *connect('unsqueeze', 'op_output')],
+                                    {'input_unsqueezed': {'shape': shape_array([1, 3, 10])},
+                                     'input_unsqueezed_d': {'shape': [1, 3, 10]},
+                                     'squeeze_dim': {'value': int64_array([0])},
+                                     'unsqueeze_dim': {'value': int64_array([1])},
+                                     }, nodes_with_edges_only=True)
+
+        flag, resp = compare_graphs(
+            sub_graph, sub_graph_ref, 'op_output', check_op_attrs=True)
+        self.assertTrue(flag, resp)
+
+        ti_node = Node(graph, 'ti')
+        ti_node.body = None  # compare main graph without body
+        flag, resp = compare_graphs(
+            graph, graph_ref, 'output', check_op_attrs=True)
```

Comment:
can't we check main graph and body graph simultaneously?
Not clear why the body graph is set to `None`

Reply:
`compare_graphs` function can't do that. It compares graphs using `==`. Fixing it may be tricky, because to compare graphs we need output names.